test/LibManagement.cpp

```diff
@@ -167,6 +167,46 @@ TEST_CASE("Valid manifests processing", "[Plugin]") {
     }
 }
 
+TEST_CASE("Valid download caching", "[Plugin]") {
+    smce::Toolchain tc{SMCE_PATH};
+    REQUIRE(!tc.check_suitable_environment());
+
+    if (std::filesystem::exists(CACHED_DOWNLOADS_PATH)) {
+        [[maybe_unused]] std::error_code ec_1;
+        std::filesystem::remove_all(CACHED_DOWNLOADS_PATH, ec_1);
+    }
+
+    std::vector<std::string> uris = {"https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.2.zip",
+                                     "https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.1.zip",
+                                     "https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.2.zip"};
+
+    std::vector<std::string> versions = {"0.2", "0.1", "0.2"};
+    std::vector<smce::PluginManifest> plugins;
+
+    for (int i = 0; i < uris.size(); i++) {
+        std::vector<smce::PluginManifest> plugins{
+            smce::PluginManifest{.name = "ESP32_AnalogWrite",
+                                 .version = versions[i],
+                                 .uri = uris[i],
+                                 .defaults = smce::PluginManifest::Defaults::none}};
+
+        smce::SketchConfig skc{.fqbn = "arduino:avr:nano", .plugins = std::move(plugins)};
+
+        smce::Sketch sk{SKETCHES_PATH "noop", std::move(skc)};
+
+        const auto ec = tc.compile(sk);
+        if (ec)
+            std::cerr << tc.build_log().second;
+
+        REQUIRE_FALSE(ec);
+    }
+
+    [[maybe_unused]] std::error_code ec_2;
```

Comment:
This should probably be checked by a `REQUIRE_FALSE`

Reply:
And scoped, so there is no need for the `_2` suffix

Reply:
i fix